src/core/manager-dump.h

```diff
@@ -5,8 +5,24 @@
 
 #include "manager.h"
 
-void manager_dump_jobs(Manager *s, FILE *f, const char *prefix);
-void manager_dump_units(Manager *s, FILE *f, const char *prefix);
-void manager_dump(Manager *s, FILE *f, const char *prefix);
-int manager_get_dump_string(Manager *m, char **ret);
+void manager_dump_jobs_by_patterns(Manager *s, FILE *f, char **patterns, const char *prefix);
+void manager_dump_units_by_patterns(Manager *s, FILE *f, char **patterns, const char *prefix);
+void manager_dump_by_patterns(Manager *s, FILE *f, char **patterns, const char *prefix);
+int manager_get_dump_string_by_patterns(Manager *m, char **patterns, char **ret);
 void manager_test_summary(Manager *m);
+
+static inline void manager_dump_jobs(Manager *s, FILE *f, const char *prefix) {
+        manager_dump_jobs_by_patterns(s, f, NULL, prefix);
+}
+
+static inline void manager_dump_units(Manager *s, FILE *f, const char *prefix) {
+        manager_dump_units_by_patterns(s, f, NULL, prefix);
+}
+
+static inline void manager_dump(Manager *m, FILE *f, const char *prefix) {
+        manager_dump_by_patterns(m, f, NULL, prefix);
+}
+
+static inline int manager_get_dump_string(Manager *m, char **ret) {
+        return manager_get_dump_string_by_patterns(m, NULL, ret);
+}
```

Comment:
Still I do not understand why you want to keep manager_dump() or friends as is.
Why not simply add new argument `char **patterns` to `manager_dump()`?

Reply:
Yeah, please keep the implementation simple.